float64[] training_query
float64[] training_answer
